### PR TITLE
deps: upgrade googletest to 1.10.0

### PR DESCRIPTION
**Pull request by derekargueta**
Description: upgrade googletest to 1.10. Also updates gen_sh_test_runner.sh to not generate test files when the given source input is empty because googletest 1.10 [introduces a static compile-time check](https://github.com/google/googletest/commit/4105eb726341fb79a2f5fbb8bba29e8d1165169b) that the test name is not empty, which is a normal case for MacOS hot restart integration tests because hot restart is disabled on macos.
Risk Level: low
Testing: existing
Docs Changes: n/a
Release Notes: n/a

Signed-off-by: Derek Argueta <dereka@pinterest.com>

### CODE REVIEW

**Comment by derekargueta** [comment]
macos seemed to fail on something unrelated
```
bazel-out/darwin-fastbuild/bin/test/integration/hotrestart_test_test_runner.cc:5:1: error: static_assert failed due to requirement 'sizeof ("") > 1' "test_name must not be empty"
TEST(ShTest, ) {
^~~~~~~~~~~~~~
```

For some reason `${TEST_NAME}` did not get set correctly in the `gen_sh_test_runner.sh` template
```
TEST(ShTest, ${TEST_NAME}) {
  Envoy::TestEnvironment::exec({${EXEC_ARGS}});
}
```

File: test/common/http/conn_manager_utility_test.cc [hunk 1]
```diff
@@ -35,7 +35,7 @@ namespace Http {
 
 class MockInternalAddressConfig : public Http::InternalAddressConfig {
 public:
-  MOCK_CONST_METHOD1(isInternalAddress, bool(const Network::Address::Instance&));
+  MOCK_METHOD(bool, isInternalAddress, (const Network::Address::Instance&), (const));
```
Comment (lizan) [inline test/common/http/conn_manager_utility_test.cc]: New `MOCK_METHOD` support `override` as well, do you want to do that in this PR or in a follow up?

Comment (derekargueta) [reply test/common/http/conn_manager_utility_test.cc]: yeah I was planning on doing that in a separate change

**Comment by derekargueta** [comment]
something about this change is consistently breaking macos build. investigating...

**Comment by derekargueta** [comment]
@dio @lizan ready for review. I updated the PR description to explain the change to `bazel/gen_sh_test_runner.sh`

**Pull request merged**